client/state/data-layer/wpcom/read/recommendations/sites/index.js

```diff
@@ -44,22 +42,24 @@ export const fromApi = response => {
 	} ) );
 };
 
-export const receiveRecommendedSitesResponse = ( store, action, response ) => {
+export const receiveRecommendedSitesResponse = ( action, response ) => {
 	if ( ! response.sites ) {
 		return;
 	}
 
-	store.dispatch(
-		receiveRecommendedSites( {
-			sites: fromApi( response ),
-			seed: action.payload.seed,
-			offset: action.payload.offset,
-		} )
-	);
+	return receiveRecommendedSites( {
+		sites: fromApi( response ),
```

Comment:
if we pass `fromApi` into `dispatchRequestEx()` then it will transform the response before calling `onSuccess`. additionally, if `fromApi` throws an exception for any reason (such as in parsing the raw data) then `onFailure` gets called instead to indicate that we couldn't parse the response